### PR TITLE
Added highlighting of installed casks to cask search

### PR DESCRIPTION
**Pull request by nath**
In response to https://github.com/caskroom/homebrew-cask/issues/30583

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by reitermarkus** [comment]
There was a [Pull Request template](https://github.com/Homebrew/brew/blob/master/.github/PULL_REQUEST_TEMPLATE.md), please follow it and run `brew tests`.

**Comment by OranShuster** [comment]
it seems the tests are failing where there is no search term

**Comment by nath** [comment]
I think the error is that it takes to long when running an empty search, as it is checking every cask for if it's installed. I still get a listing of all the casks, but it takes forever.

I'll see if there's a faster way to check if a cask is installed.

File: Library/Homebrew/cask/lib/hbc/utils.rb [hunk 1]
```diff
@@ -36,6 +36,13 @@ def odebug(title, *sput)
   puts sput unless sput.empty?
 end
 
+def highlight_installed(token)
+  if (Hbc.caskroom/token).directory?
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/utils.rb]: Try `Cask.new(token).installed?`, so we still have the same `#installed?` logic everywhere.

Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/utils.rb]: Also, move this method into `search.rb`.

Comment (nath) [inline Library/Homebrew/cask/lib/hbc/utils.rb]: Done, thanks for being so helpful with this!

File: Library/Homebrew/cask/lib/hbc/cli/search.rb [hunk 2]
```diff
@@ -49,7 +49,15 @@ def self.render_results(exact_match, partial_matches, search_term)
         else
           ohai "Partial matches"
         end
-        puts Formatter.columns(partial_matches)
+        highlighted = partial_matches.map { |match| highlight_installed match }
+        puts Formatter.columns(highlighted)
+      end
+
+      def self.highlight_installed(token)
+        if Hbc::Cask.new(token).installed?
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/cli/search.rb]: You can leave off the `Hbc::` here.

File: Library/Homebrew/cask/lib/hbc/cli/search.rb [hunk 3]
```diff
@@ -49,7 +49,15 @@ def self.render_results(exact_match, partial_matches, search_term)
...
-        puts Formatter.columns(partial_matches)
+        highlighted = partial_matches.map { |match| highlight_installed match }
+        puts Formatter.columns(highlighted)
+      end
+
+      def self.highlight_installed(token)
+        if Cask.new(token).installed?
+          token = pretty_installed token
+        end
+        token
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/cli/search.rb]: Make this:

```ruby
return token unless Cask.new(token).installed?
pretty_installed token
```

File: Library/Homebrew/cask/lib/hbc/cli/search.rb [hunk 4]
```diff
@@ -49,7 +49,15 @@ def self.render_results(exact_match, partial_matches, search_term)
         else
           ohai "Partial matches"
         end
-        puts Formatter.columns(partial_matches)
+        highlighted = partial_matches.map { |match| highlight_installed match }
```
Comment (reitermarkus) [inline Library/Homebrew/cask/lib/hbc/cli/search.rb]: `partial_matches.map(&method(:highlight_installed))`

File: Library/Homebrew/test/cask/cli/search_spec.rb [hunk 5]
```diff
@@ -56,4 +56,16 @@
       Hbc::CLI::Search.run("caskroom")
     }.to output(/^No Cask found for "caskroom"\.\n/).to_stdout
   end
+
+  it "doesn't highlight not installed packages" do
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/cli/search_spec.rb]: This should be ”non-installed packages“, which still doesn't sound right, so maybe just ”packages that aren't installed“?

**Comment by reitermarkus** [comment]
Thank you for your contribution!

**Pull request merged**